### DISCUSSION TITLE
Selfhosted VM runner diskspace

### DISCUSSION THREAD

**Discussion by CarlOhlsson** [microsoft/AL-Go#473] [Q&A] [answered]
Hi!

We are using selfhosted VM runners and they are working great but after some time they start running out of disk space. I doubled the available disk space and my guess is that it will work for another 2-3 months.

I don’t think this is a good solution and would rather like to remove the unused files since my guess is that it is old versions of BC used by Docker? My knowledge of Docker is very limited and last time I tried to clean the data by googling I broke everything and ended up having to create a new VM. 

See the image I have attached with what is taking up the space of my disk. There must be others with the same issue as me, have you solved this in some way?

![image](https://user-images.githubusercontent.com/11760980/232011533-686d5ac7-b997-4a15-acf0-bc1b5d934818.png)

Edit: Think this is related to issue #459

**Comment by AskeHolst** [comment]
I got the same issue on our two runner VMs (as mentioned in #459). 
Running **docker system prune -a -f** solved the immediate problem and freed up almost all space on one of the runners. On the other, it freed up a lot, but there is still 50GB in the d folder. They are otherwise identical. but one was left with 50GB more free space than the other. I have no idea why.

**Comment by freddydk** [comment, marked as answer]
With the changes to BcContainerHelper on July 6th as specified in #459, this should hopefully not be a problem anymore
